src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -286,15 +292,20 @@ private ModuleCommands parseInModuleCommandsFromInput(String input) {
      * @param input full user input string
      * @return a Lesson object with the details entered by the user. Fields not found will be left as an empty string.
      */
-    private Lesson parseNewLessonDetails(String input) {
-        // TODO - validate input
+    private Lesson parseNewLessonDetails(String input) throws CommandException {
 
         // initialize an array of empty strings to store lesson details
         String[] allDetails = new String[ENTRY_LESSON_MAX_PARSER];
-        Arrays.fill(allDetails, "");
+        Arrays.fill(allDetails, PLACEHOLDER);
 
         // to remove only the first two words "add lesson"
-        String[] lessonDetails = input.trim().split(" ", 3);
+        String[] lessonDetails = input.trim().split(WHITESPACE, 3);
+
+        // ERROR - User does not enter any parameters.
+        if (lessonDetails.length < 3) {
```

Comment:
Preferable to change the magic number to a constant

Reply:
Magic strings are as placeholders for now.